docs: distinguish between `$fetch` and fetch composables (#21228)
nuxt/nuxt: docs/3.api/1.composables/use-async-data.md

Within your pages, components, and plugins you can use useAsyncData to get access to data that resolves asynchronously.

::alert{type=warning}

> `useAsyncData` is a composable meant to be called directly in a setup function and only there. It returns reactive composables and handles adding responses to the Nuxt payload so they can be passed from server to client without re-fetching the data on client side when the page hydrates.

Reviewer: ```suggestion
`useAsyncData` is a composable meant to be called directly in a setup function, plugin, or route middleware. It returns reactive composables and handles adding responses to the Nuxt payload so they can be passed from server to client without re-fetching the data on client side when the page hydrates.
```
